openkfw/TruBudget
# api, frontend: autocheck file integrity

### Checklist




### Description


### How to test

0. You will need to replace a file directly on storage, I tested with azurite and azure storage explorer
1. Start TB in full mode, provisioned
2. Amazonas Fund > Furniture has a workflow item with two files, both of which can be downloaded and have correct hash
3. Using azure storage explorer or other tool, note down the file metadata
4. Using the same tool, delete a file. Upload a different file with the same name and set metadata to previous values
5. Using TB UI, there should now be a warning about incorrect hash and a log in api console
<img width="1120" alt="Screenshot 2024-08-15 at 09 53 46" src="https://github.com/user-attachments/assets/779c3644-376d-4b89-9d7f-2055a6172819">




Closes  #1924
